test/agent.js

```diff
@@ -186,138 +186,121 @@ test('#addTags', function (t) {
 })
 
 test('#addFilter() - invalid argument', function (t) {
-  t.plan(5)
-  APMServer()
+  t.plan(3 + APMServerWithDefaultAsserts.asserts)
+  APMServerWithDefaultAsserts(t, {}, {expect: 'error'})
     .on('listening', function () {
-      this.agent.addFilter(function (data) {
-        var error = data.errors[0]
-        t.equal(++error.context.custom.order, 1)
-        return data
+      this.agent.addFilter(function (obj) {
+        t.equal(++obj.context.custom.order, 1)
+        return obj
       })
       this.agent.addFilter('invalid')
-      this.agent.addFilter(function (data) {
-        var error = data.errors[0]
-        t.equal(++error.context.custom.order, 2)
-        return data
+      this.agent.addFilter(function (obj) {
+        t.equal(++obj.context.custom.order, 2)
+        return obj
       })
       this.agent.captureError(new Error('foo'), {custom: {order: 0}})
     })
-    .on('request', validateErrorRequest(t))
-    .on('body', function (body) {
-      t.deepEqual(body.errors[0].context.custom.order, 2)
+    .on('data-error', function (data) {
+      t.deepEqual(data.context.custom.order, 2)
       t.end()
     })
 })
 
 test('#captureError()', function (t) {
   t.test('with callback', function (t) {
-    t.plan(5)
-    APMServer()
+    t.plan(2 + APMServerWithDefaultAsserts.asserts)
+    APMServerWithDefaultAsserts(t, {}, {expect: 'error'})
       .on('listening', function () {
         this.agent.captureError(new Error('with callback'), function () {
-          t.ok(true, 'called callback')
-          t.end()
+          t.pass('called callback')
         })
       })
-      .on('request', validateErrorRequest(t))
-      .on('body', function (body) {
-        t.equal(body.errors.length, 1)
-        t.equal(body.errors[0].exception.message, 'with callback')
+      .on('data-error', function (data) {
+        t.equal(data.exception.message, 'with callback')
+        t.end()
```

Comment:
Does tape work correctly when you explicitly `t.end()` while using a `t.plan(...)`?

Reply:
Yes, all it does when reaching the `t.end` is just fail if the number of assertions doesn't match what's planned. This is a way to to check our exceptions of async execution order. I.e. we reach this point, we should have made the planned number of assertions.

Reply:
Ah, cool. 👍